app/src/main/java/com/kickstarter/ApplicationModule.java

```diff
@@ -172,6 +172,32 @@ static Environment provideEnvironment(final @NonNull @ActivitySamplePreference I
       .build();
   }
 
+  @Provides
+  @Nullable
+  @Singleton
+  static Analytics provideSegment(final @NonNull Build build, final @ApplicationContext @NonNull Context context) {
+    String apiKey = "";
+    Analytics segmentClient = null;
+
+    if (build.isRelease() && Build.isExternal()) {
+      apiKey = Secrets.Segment.PRODUCTION;
+    }
+    if (build.isDebug()) {
+      apiKey = Secrets.Segment.STAGING;
+    }
+
+    if (context instanceof KSApplication && !((KSApplication) context).isInUnitTests()) {
```

Comment:
Since we're already checking if context is an instance of KSApplication, do we need to cast? Or would that be a compiler error?

Reply:
Compilation error, it needs the cast